docs/operator-manual/profiling.md

```diff
@@ -0,0 +1,38 @@
+# Profiling with `pprof`
+
+[`pprof`](https://go.dev/blog/pprof) is a powerful tool for profiling Go applications. It is part of the Go standard library and provides rich insights into the performance characteristics of Go programs, such as CPU usage, memory allocation, and contention.
+
+## Basic Usage
+
+Enable profiling endpoints by setting the environment variable `ARGO_PPROF` with your preferred port. For instance, `ARGO_PPROF=8888` will start profiling endpoints on port `8888`.
+
+`pprof` has two modes: interactive and non-interactive. Non-interactive mode generates profiling data for future analysis. Interactive mode launches a web server to visualize the profiling data in real-time.
+
+!!! Note
+    In the below commands, replace http://localhost:6060 with the appropriate URL of your Argo component.
```

Comment:
Just realized, we might want to recommend that users port-forward the `pprof` port to their `localhost`, since the port should generally not be accessible via Ingress (unlike the UI, for example)

Reply:
I used localhost to implicitly suggest the port-forward, but you are right and in this case, more explicit is better as exposing this server could be risky